src/plugins/intel_cpu/src/nodes/llm_mlp.cpp

```diff
@@ -200,7 +201,11 @@ struct LLMMLP::Impl {
     }
 
     void setM(int M) {
-        if (m_M < M) {
+        uint8_t* cur_scratch_base = nullptr;
```

Comment:
Comparing the memory pointer is ambiguous. The condition behind is that the scratch buffer isn't big enough. Could you check why the scratch buffer is not big enough ?

Reply:
The problem is not big enough scratch here, it comes from situation:
1, each mlp layer will create a Memory object using a same scratch size such as 4M, then uses the scratch pointer to initialize the class member m_actUp, m_tempC, actually because the size is same, the pointer is same too.
2, some layers such as last Matmul `lm_head` may need a bigger scratch then scratch is re-created and the pointers used in m_actUp, m_tempC become invalid.

Here use pointer to detect the condition: changed scratch.

Reply:
Does different MLP layers share same LLMMLP executor even if they have different M, K, N inside ?

Reply:
There is no executor, the jit kernel which is a [global variable](https://github.com/openvinotoolkit/openvino/blob/f5463512c6d432192fb8152819511dbc8261e540/src/plugins/intel_cpu/src/nodes/kernels/x64/mlp_kernel.hpp#L121) does not use M, N, K to generate the kernel code.